hooks/primary_publish.py

```diff
@@ -995,49 +996,65 @@ def _do_photoshop_publish(
-            "sg_last_frame": 1,
-            "entity": ctx.entity,
-            "sg_path_to_frames": publish_path,
-            "project": ctx.project,
-            "sg_task": sg_task,
-            "code": tank_publish["code"],
-            "created_by": ctx.user,
-        }
-        
-        if tank.util.get_published_file_entity_type(self.parent.tank) == "PublishedFile":
-            data["published_files"] = [tank_publish]
-        else:# == "TankPublishedFile"
-            data["tank_published_file"] = tank_publish
-        
-        version = self.parent.shotgun.create("Version", data)
-        
-        # upload jpeg
-        progress_cb(70.0, "Uploading to Shotgun...")
-        self.parent.shotgun.upload("Version", version['id'], jpg_pub_path, "sg_uploaded_movie" )
-        
         try:
-            os.remove(jpg_pub_path)
-        except:
-            pass
-        
+            # The export_as_jpeg method was not available in early releases
+            # of the tk-photoshopcc engine. It is not possible to specify a minimum
+            # release for an engine in a multi app, so check if the method is
+            # available and issue a warning if not.
+            if not hasattr(self.parent.engine, "export_as_jpeg"):
+                raise UserWarning(
+                "A more recent release than %s %s is needed to generate a Jpeg Version." % (
+                    self.parent.engine.name,
+                    self.parent.engine.version,
+                ))
+            # Export a Jpeg image
+            jpeg_pub_path = self.parent.engine.export_as_jpeg()
+
+            # Then register version
+            progress_cb(60.0, "Creating Version...")
+            ctx = self.parent.context
+            data = {
+                "user": ctx.user,
+                "description": comment,
+                "sg_first_frame": 1,
+                "frame_count": 1,
+                "frame_range": "1-1",
+                "sg_last_frame": 1,
+                "entity": ctx.entity,
+                "sg_path_to_frames": publish_path,
+                "project": ctx.project,
+                "sg_task": sg_task,
+                "code": tank_publish["code"],
+                "created_by": ctx.user,
+            }
+            
+            if tank.util.get_published_file_entity_type(self.parent.tank) == "PublishedFile":
+                data["published_files"] = [tank_publish]
+            else:# == "TankPublishedFile"
+                data["tank_published_file"] = tank_publish
+            
+            version = self.parent.shotgun.create("Version", data)
+            
+            # upload jpeg
+            progress_cb(70.0, "Uploading to Shotgun...")
+            self.parent.shotgun.upload(
+                "Version",
+                version['id'],
+                jpeg_pub_path,
+                "sg_uploaded_movie"
+            )
+            
+            try:
+                os.remove(jpeg_pub_path)
+            except:
```

Comment:
We should stay away from bare excepts. That's going to catch system-level exceptions that do not inherit from Exception that should never be caught by client code. Stuff like keyboard interrupts and SIGTERMs.

Best to either except Exception or, better yet in this case, except OSError.